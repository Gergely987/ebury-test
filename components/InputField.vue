<template>
    <div class="input-field field">
        <b-field :type="validity || !dirty ? '' : 'is-danger'">
            <b-input 
                v-model="value" 
                :placeholder="config.placeholder"
                :maxlength="config.maxlength" 
                :type="config.type"
                @blur="validateField"
            ></b-input>
        </b-field>
    </div>
</template>

<script>
export default {
    name: 'InputField',
    props: {
        config: {
            type: Object,
            required: true
        }
    },
    data() {
        return {
            value: '',
            validity: !this.config.required,
            dirty: false
        }
    },
    methods: {
        validateField() {
            this.dirty = true;

            if (this.config.required && !this.value.length) {
                this.validity = false;
            } else if(this.config.validation === 'email' && this.value.length) {

                const emailArray = this.value.split(',');
                let valid = true;

                for(let i = 0; i < emailArray.length; i++) {
                    valid = valid && this.validEmail(emailArray[i].replace(/\s/g, ''));
                }
                this.validity = valid;

            } else {
                this.validity = true;
            }

            this.$store.dispatch("setValidation", { name: this.config.name, validity: this.validity });

            this.$store.dispatch("setValue", { name: this.config.name, value: this.value });

            this.$store.dispatch("testFormValidity", this.$store.state.emailData);
           
        },
        validEmail(email) {
            // eslint-disable-next-line no-useless-escape
            const rule = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
            return rule.test(email);
        }
    }
}
</script>

<style lang="less">
@import "../assets/css/variables.less";

.input-field {
    width: 100%;
    margin-bottom: 15px;

    input {
        height: 40px;
    }

    .help {
      display: none;
    }

    .textarea {
        min-height: 200px;
    }
}

</style>