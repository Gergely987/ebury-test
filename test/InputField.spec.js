import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import InputField from '@/components/InputField.vue'

Vue.config.ignoredElements = ['b-input', 'b-field']


describe('InputField', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(InputField)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('matches input snapshot', () => {
    const config = {
        name: 'recipienctto',
        placeholder: 'To',
        required: true
    };
    const wrapper = shallowMount(InputField, {
        propsData: { config }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches textarea snapshot', () => {
    const config = {
        name: 'message',
        required: true,
        type: 'textarea'
    };
    const wrapper = shallowMount(InputField, {
        propsData: { config }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})