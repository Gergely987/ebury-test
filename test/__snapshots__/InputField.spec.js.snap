// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField matches input snapshot 1`] = `"<div class=\\"input-field field\\"><b-field type=\\"\\"><b-input></b-input></b-field></div>"`;

exports[`InputField matches textarea snapshot 1`] = `"<div class=\\"input-field field\\"><b-field type=\\"\\"><b-input></b-input></b-field></div>"`;
